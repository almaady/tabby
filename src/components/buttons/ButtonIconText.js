import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import Icon from "../icons/Icon";

/**
 * @class ButtonText - Is a button that only contains text
 */
class ButtonIconText extends React.PureComponent {
  /**
   * @property {string} text -The text of the button
   * @property {func} onClick - The function triggered by the button
   * @property {string} className - Just in case you need another class
   * @property {string} color -  The color of the button
   * @property {string} type - Button type
   * @property {string} icon - Name of the icon
   * @property {string} iconColor - Color of the icon
   * @property {string} iconSize - Size of the icon, by default is 16px
   * @property {bool} iconRight - to define the position of the icon
   * @property {bool} disabled - Button could be disabled
   * @property {string} borderColor - Button could have a border color
   * @see See './../../../styles/variables/_colors.scss' for the list of color names
   */
  static propTypes = {
    text: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
    color: PropTypes.string.isRequired,
    className: PropTypes.string,
    type: PropTypes.string,
    icon: PropTypes.string.isRequired,
    iconColor: PropTypes.string.isRequired,
    iconSize: PropTypes.string,
    iconRight: PropTypes.bool,
    disabled: PropTypes.bool,
    borderColor: PropTypes.string
  };


  render() {
    const { color, onClick, text, className, type, iconColor, icon, iconSize, iconRight, disabled, borderColor } = this.props;
    return (
      <button
        className={classNames("btn", "btn__icon--text", `bg--${color}`, { "btn__icon--right": iconRight}, `border--${borderColor}`, className)}
        onClick={onClick}
        type={type}
        disabled={disabled}
      >
        <Icon icon={icon} color={iconColor} size={iconSize}/>
        <p>{text}</p>
      </button>
    );
  }
}
ButtonIconText.defaultProps = {
  className: "",
  type: "button",
  iconSize: "R",
  iconRight: false,
  disabled: false,
  borderColor: ""
};

export default ButtonIconText;
