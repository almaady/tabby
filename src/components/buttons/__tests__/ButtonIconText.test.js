import React from "react"
import {shallow, mount} from "enzyme"
import sinon from "sinon"

import ButtonIconText from "../ButtonIconText"

describe("ButtonIconText", () => {
  const action = sinon.stub()
  let props;
  beforeEach(() => {
    props = {
      text:'text',
      onClick: action,
      color: "blue",
      icon: "down",
      iconColor: "white"
    };
  });

  it("should render correctly with default state", () => {
    const component = shallow(<ButtonIconText {...props} />);
    expect(component).toMatchSnapshot();
  });
  it("should call action whe  you click", () => {
    const component = shallow(<ButtonIconText {...props} />);
    component.simulate("click");
    expect(action.called).toBeTruthy();
  });
  it("should render new className", () => {
    const component = shallow(<ButtonIconText {...props} className="btn-2" />);
    expect(component.hasClass("btn-2"));
  });
  it("should has an icon", () => {
    const component = mount(<ButtonIconText {...props} icon="cross" />);
    expect(component.find("Icon")).toBeTruthy();
  });
});
