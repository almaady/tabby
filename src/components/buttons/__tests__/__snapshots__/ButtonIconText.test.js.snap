// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonIconText should render correctly with default state 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonIconText
    borderColor=""
    className=""
    color="blue"
    disabled={false}
    icon="down"
    iconColor="white"
    iconRight={false}
    iconSize="R"
    onClick={[Function]}
    text="text"
    type="button"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Icon
          color="white"
          icon="down"
          size="R"
        />,
        <p>
          text
        </p>,
      ],
      "className": "btn btn__icon--text bg--blue border--",
      "disabled": false,
      "onClick": [Function],
      "type": "button",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "color": "white",
          "icon": "down",
          "size": "R",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "text",
        },
        "ref": null,
        "rendered": "text",
        "type": "p",
      },
    ],
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Icon
            color="white"
            icon="down"
            size="R"
          />,
          <p>
            text
          </p>,
        ],
        "className": "btn btn__icon--text bg--blue border--",
        "disabled": false,
        "onClick": [Function],
        "type": "button",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": "white",
            "icon": "down",
            "size": "R",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "text",
          },
          "ref": null,
          "rendered": "text",
          "type": "p",
        },
      ],
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
